### PR TITLE
BUG: Fix Kwarg only in update_env

### PR DESCRIPTION
**Pull request by Carreau**
closes jupyterlab/jupyterlab#15301

### CODE REVIEW

**Comment by blink1073** [comment]
Looks like there's still an issue with the downstream server tests.

```python
   Traceback (most recent call last):
    File "/home/runner/test_venv/lib/python3.11/site-packages/tornado/web.py", line 1786, in _execute
      result = await result
               ^^^^^^^^^^^^
    File "/home/runner/test_venv/downstream_test/jupyter_server/jupyter_server/services/sessions/handlers.py", line 172, in patch
      await sm.update_session(session_id, **changes)
    File "/home/runner/test_venv/downstream_test/jupyter_server/jupyter_server/services/sessions/sessionmanager.py", line 469, in update_session
      self.kernel_manager.update_env(kernel_id=kernel_id, env=self.get_kernel_env(path, name))
    File "/home/runner/test_venv/lib/python3.11/site-packages/jupyter_client/multikernelmanager.py", line 226, in update_env
      self._kernels[kernel_id].update_env(env=env)
    File "/home/runner/test_venv/lib/python3.11/site-packages/jupyter_client/manager.py", line 284, in update_env
      self._launch_args['env'].update(env)
      ~~~~~~~~~~~~~~~~~^^^^^^^
  KeyError: 'env'
```

**Comment by blink1073** [comment]
I'm reverting this change in server for now.

**Comment by Carreau** [comment]
I pushed a check that env exists.

**Comment by blink1073** [comment]
Have you tested this with server to 2.8.1?

**Comment by Carreau** [comment]
Ok, I've been more agressive on the condition, checking that everything exists and are dicts, and added all the corresponding test cases.

I've tested on Jupyter_server  2.7.0, 2.9.1 (does not crash, but does nothing), 
and on 2.9.0 (it does work an update `__session__`).

**Comment by Carreau** [comment]
Let me see if I can fix the typing of `_launch_args`, that is set to `Any`, I think we can make it t.Optional[Dict]

**Comment by Carreau** [comment]
ok, I can't make it ` t.Optional[Dict[str, Any]]`, as `Dict` is not subscriptable, but I did make it `t.Optional[Dict]`

File: jupyter_client/manager.py [hunk 1]
```diff
@@ -314,7 +320,7 @@ def format_kernel_cmd(self, extra_arguments: t.Optional[t.List[str]] = None) ->
 
         if self.kernel_spec:  # type:ignore[truthy-bool]
             ns["resource_dir"] = self.kernel_spec.resource_dir
-
+        assert self._launch_args is not None
```
Comment (Carreau) [inline jupyter_client/manager.py:323]: This helps mypy, the line below would fail anyway if it was None.

Comment (blink1073) [reply jupyter_client/manager.py:323]: | ok, I can't make it  t.Optional[Dict[str, Any]], as Dict is not subscriptable

You can use  `t.Optional["Dict[str, Any]"]` to get around the runtime issue.

Comment (Carreau) [reply jupyter_client/manager.py]: Ah ! True, let's try

**Comment by blink1073** [comment]
Papermill failure is known: https://github.com/nteract/papermill/issues/737

**Review by blink1073 — APPROVED** [review]
Thanks!

**Pull request merged**